src/mame/heathkit/tlb.h

```diff
@@ -190,4 +214,61 @@ DECLARE_DEVICE_TYPE(HEATH_SUPER19, heath_super19_tlb_device)
 DECLARE_DEVICE_TYPE(HEATH_WATZ, heath_watz_tlb_device)
 DECLARE_DEVICE_TYPE(HEATH_ULTRA, heath_ultra_tlb_device)
 
+
+/**
+ * Connector for the Terminal Logic Board in an H-89 class computer
+ */
+class heath_tlb_connector : public device_t,
+							public device_single_card_slot_interface<device_heath_tlb_card_interface>
+{
+public:
+
+	template <typename T>
+	heath_tlb_connector(const machine_config &mconfig, const char *tag, device_t *owner, T &&opts, const char *dflt, bool fixed = false) :
+		 heath_tlb_connector(mconfig, tag, owner, 0)
+	{
+		option_reset();
+		opts(*this);
+		set_default_option(dflt);
+		set_fixed(fixed);
+	}
+
+	heath_tlb_connector(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock = 0);
+	virtual ~heath_tlb_connector();
+
+	// computer interface
+	auto serial_data_callback() { return m_write_sd.bind(); }
+	auto dtr_callback() { return m_dtr_cb.bind(); }
+	auto rts_callback() { return m_rts_cb.bind(); }
+	auto reset_cb() { return m_reset.bind(); }
+
+	// card interface
+	void serial_in_w(int state) { if (m_tlb) m_tlb->serial_in_w(state); };
+	void rlsd_in_w(int state) { if (m_tlb) m_tlb->rlsd_in_w(state); };
+	void dsr_in_w(int state) { if (m_tlb) m_tlb->dsr_in_w(state); };
+	void cts_int_w(int state) { if (m_tlb) m_tlb->cts_int_w(state); };
+
+	void serial_out_b(int data) { m_write_sd(data); };
+	void dtr_out(int data)  { m_dtr_cb(data); };
+	void rts_out(int data)  { m_rts_cb(data); };
+	void reset_out(int data) { m_reset(data); };
+
+protected:
+	virtual void device_start() override;
+
+	devcb_write_line m_write_sd;
+	devcb_write_line m_dtr_cb;
+	devcb_write_line m_rts_cb;
+	devcb_write_line m_reset;
+
+	device_heath_tlb_card_interface *m_tlb;
+};
+
+
+// device type definition
+DECLARE_DEVICE_TYPE(HEATH_TLB_CONNECTOR, heath_tlb_connector)
+
+extern template class device_finder<heath_tlb_connector, false>;
+extern template class device_finder<heath_tlb_connector, true>;
```

Comment:
I wasn't sure what the lines were for, and I think I did pull this from a2bus, since it was at the top of the list as I was searching for examples to see how to make this into a slot device. I'll remove them.

Reply:
The only device header I can think of off the top of my head that does something like this in the header is the floppy drive connector abstraction.  It’s unorthodox in several ways.